dask/dataframe/core.py

```diff
@@ -140,6 +138,17 @@ class _Frame(Base):
     _default_get = staticmethod(threaded.get)
     _finalize = staticmethod(finalize)
 
+    def __new__(cls, dsk, _name, metadata, divisions):
+        if isinstance(metadata, (Series, pd.Series)):
+            metadata = metadata.name
+        elif isinstance(metadata, (DataFrame, pd.DataFrame)):
+            metadata = metadata.columns
```

Comment:
Eventually it would be nice to document the constructor behavior somehow.  Perhaps not yet (things are changing rapidly) but we should consider this once things settle a bit.